utils/src/duration.rs

```diff
@@ -0,0 +1,26 @@
+use std::time::Duration;
+
+/// Extension trait to add methods to `Duration`
+pub trait Utils {
```

Comment:
nit: not in love with this trait name. Maybe we call it `Extension` or `U64` (don't love those either but `Utils` just feels coarse).